src/main/java/teammates/logic/core/FeedbackResponsesLogic.java

```diff
@@ -838,4 +983,196 @@ private void addNewResponses(
 
         return viewableResponses;
     }
+    
+    public boolean isResponseValid(FeedbackResponseAttributes response,
+                                   FeedbackQuestionAttributes question) {
+        if (response == null || question == null) {
+            return false;
+        }
+        return response.isValid() && isResponseGiverAndReceiverValid(response, question);
+    }
+
+    public boolean isResponseGiverAndReceiverValid(FeedbackResponseAttributes response,
+                                                   FeedbackQuestionAttributes question) {
+        if (question == null || response == null) {
+            return false;
+        }
+
+        return isResponseGiverTypeMatchedToQuestionSetting(response, question) 
+               && isResponseRecipientTypeMatchedToQuestionSetting(response, question);
```

Comment:
Here it is. This part is adding a few database reads every time a new instructor is created (like #4818, but on a lighter scale). Any way they can be optimized? If not, then adding the timeout at `test.template.properties` to 20-25 seconds should be enough to make it pass consistently.